File: 预训练模型/支持向量机/py/api/posts.py
import requests

# 查询帖子信息列表
def list_posts(query):
    url = '/posts/posts/list'
    try:
        response = requests.get(url, params=query)
        response.raise_for_status()  # 若请求失败（状态码非 200），抛出异常
        return response.json()
    except requests.RequestException as e:
        print(f"请求出错: {e}")
        return None

# 修改帖子信息
def update_posts(data):
    url = '/posts/posts'
    try:
        response = requests.put(url, json=data)
        response.raise_for_status()  # 若请求失败（状态码非 200），抛出异常
        return response.json()
    except requests.RequestException as e:
        print(f"请求出错: {e}")
        return None
    

File: YunJian-App/api/posts/posts.js
import request from '@/utils/request'

// 查询帖子信息列表
export function listPosts(query) {
  return request({
    url: '/posts/posts/list',
    method: 'get',
    params: query
  })
}

// 查询帖子信息详细
export function getPosts(id) {
  return request({
    url: '/posts/posts/' + id,
    method: 'get'
  })
}

// 新增帖子信息
export function addPosts(data) {
  return request({
    url: '/posts/posts',
    method: 'post',
    data: data
  })
}

// 修改帖子信息
export function updatePosts(data) {
  return request({
    url: '/posts/posts',
    method: 'put',
    data: data
  })
}

// 删除帖子信息
export function delPosts(id) {
  return request({
    url: '/posts/posts/' + id,
    method: 'delete'
  })
}
